python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -306,6 +200,192 @@ def sensor_fn(context):
     )
 
 
+def reconcile(
+    repository_def,
+    asset_selection: AssetSelection,
+    instance: "DagsterInstance",
+    wait_for_in_progress_runs: bool,
+    wait_for_all_upstream: bool,
+    latest_consumed_storage_ids_by_asset_key_str: Mapping[str, int],
+    run_tags: Mapping[str, str],
+) -> Tuple[Sequence[RunRequest], Mapping[str, int]]:
+    asset_defs_by_key = repository_def._assets_defs_by_key  # pylint: disable=protected-access
+    source_asset_defs_by_key = (
+        repository_def.source_assets_by_key  # pylint: disable=protected-access
+    )
+    upstream: Mapping[AssetKey, Set[AssetKey]] = _get_upstream_mapping(
+        selection=asset_selection,
+        assets=asset_defs_by_key.values(),
+        source_assets=source_asset_defs_by_key.values(),
+    )
+
+    should_materialize: Set[AssetKey] = set()
+    newly_consumed_storage_ids_by_asset_key_str: Dict[str, int] = {}
+
+    # sort the assets topologically so that we process them in order
+    toposort_assets = list(toposort.toposort(upstream))
+    # unpack the list of sets into a list and only keep the ones we are monitoring
+    toposort_assets = [
+        asset for layer in toposort_assets for asset in layer if asset in upstream.keys()
+    ]
+
+    instance_queryer = CachingInstanceQueryer(instance)
+
+    # determine which assets should materialize based on the materialization status of their
+    # parents
+    for current_asset_key in toposort_assets:
+        current_asset_cursor = latest_consumed_storage_ids_by_asset_key_str.get(
+            str(current_asset_key)
+        )
+
+        parent_update_records = _get_parent_updates(
+            current_asset=current_asset_key,
+            parent_assets=upstream[current_asset_key],
+            cursor=current_asset_cursor,
+            will_materialize_set=should_materialize,
+            wait_for_in_progress_runs=wait_for_in_progress_runs,
+            instance_queryer=instance_queryer,
+        )
+
+        condition = all if wait_for_all_upstream else any
+        if condition(
+            materialization_status for materialization_status, _ in parent_update_records.values()
+        ):
+            should_materialize.add(current_asset_key)
+
+            # get the cursor value by selecting the max of all the candidates. If we're using a
+            # sharded event log storage, compare timestamps, otherwise compare storage ids. See
```

Comment:
comment about sharded event log storage can be removed

Reply:
Thanks for catching - fixed